etcd/v3/kv_etcd.go

```diff
@@ -803,43 +807,63 @@ func (et *etcdKV) watchStart(
 	if waitIndex != 0 {
 		opts = append(opts, e.WithRev(int64(waitIndex+1)))
 	}
+	session, err := concurrency.NewSession(et.kvClient, concurrency.WithTTL(defaultSessionTimeout))
```

Comment:
Yes I had that cancel context, but cancelling it does not have any effect on the go-routine which is handling the watch responses (watchChan). Will add it back.

Should we refresh the kvClient in the watch api ?

The v2 docs say that the client goes to the next etcd - https://github.com/coreos/etcd/tree/master/client#caveat
But I could not find any doc or mention for clientv3.